File: prob03/node/app/prob3.js
const number = 600851475143

function get_prime_factors(num) {
  var factors = [];
  var current = 2;
  while(current <= num && num != 1) {
    if(num % current == 0) {
      factors.push(current);
      num /= current;
    }
    else {
      current ++;
    }
  }
  return factors;
}

factors = get_prime_factors(600851475143);
console.log(factors[factors.length - 1]);
return factors[factors.length - 1];

File: prob03/golang/prob3.go
package main

import "fmt"

func main() {
  fmt.Println(largest_prime_factor(600851475143));
}

func largest_prime_factor(num int) int {
  factors := []int{};
  current := 2;

  for ;current <= num && num != 1; {
    if num % current == 0 {
      factors = append(factors,current);
      num /= current;
    } else {
      current ++;
    }
  }
  return factors[len(factors)-1];
}
